pandas/io/common.py

```diff
@@ -135,7 +142,7 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None):
 
     if _is_url(filepath_or_buffer):
         req = _urlopen(str(filepath_or_buffer))
```

Comment:
You can also do `compression = req.headers['Content-Encoding']` here, if it's being served as a gzip encoded.